File: this_is_coding_test/그리디 알고리즘/거스름돈.py
n = 1260
count = 0

array = [500, 100, 50, 10]
for coin in array: # 화폐의 갯수만큼 반복문 실행
    count += n // coin 
    # 거스름돈을 coin으로 나눈 몫이 코인의 갯수, 그것을 count로 더한다
    n %= coin # 거스름돈을 coin으로 나눈 나머지를 다시 새로운 거스름돈으로 설정

print(count)

File: this_is_coding_test/그리디 알고리즘/거스름 돈.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        int n, count = 0;
        Scanner s = new Scanner(System.in);

        n = Integer.parseInt(s.next()); // 거슬러 줘야할 금액 입력받기
        int[] Coin = new int[]{500, 100, 50, 10};

        for (int i = 0; i < 4; i++) {
            count += n / Coin[i];
            n %= Coin[i];
        }
        System.out.println(count);
    }
}
